sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRepositoryAsyncClient.java

```diff
@@ -0,0 +1,650 @@
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<TagProperties> listTags() {
+        return listTags(null);
+    }
+
+    /**
+     * List tags of a repository.
+     *
+     * @param options tagOptions to be used for the given operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws ResourceNotFoundException thrown if the given digest was not found.
+     * @return list of tag details.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<TagProperties> listTags(ListTagsOptions options) {
+        return new PagedFlux<>(
+            (pageSize) -> withContext(context -> listTagsSinglePageAsync(pageSize, options, context)),
+            (token, pageSize) -> withContext(context -> listTagsNextSinglePageAsync(token, context)));
+    }
+
+    Mono<PagedResponse<TagProperties>> listTagsSinglePageAsync(Integer pageSize, ListTagsOptions options, Context context) {
+        try {
+            if (pageSize != null && pageSize < 0) {
+                return monoError(logger, new IllegalArgumentException("'pageSize' cannot be negative."));
+            }
+
+            if (context == null) {
+                return monoError(logger, new NullPointerException("'context' cannot be null."));
+            }
+
+            String orderBy = null;
+            if (options != null && options.getTagOrderBy() != null) {
+                orderBy = options.getTagOrderBy().toString();
+            }
+
+            return this.serviceClient.getTagsSinglePageAsync(repositoryName, null, pageSize, orderBy, null, context)
+                .map(res -> Utils.getPagedResponseWithContinuationToken(res, this::getTagProperties))
+                .onErrorMap(Utils::mapException);
+        } catch (RuntimeException e) {
+            return monoError(logger, e);
+        }
+    }
+
+    private List<TagProperties> getTagProperties(List<TagAttributesBase> baseValues)
+    {
+        Objects.requireNonNull(baseValues);
+
+        return baseValues.stream().map(value -> new TagProperties(
+            value.getName(),
+            repositoryName,
+            value.getDigest(),
+            value.getWriteableProperties(),
+            value.getCreatedOn(),
+            value.getLastUpdatedOn()
+        )).collect(Collectors.toList());
+    }
+
+    Mono<PagedResponse<TagProperties>> listTagsNextSinglePageAsync(String nextLink, Context context) {
+        try {
+            return this.serviceClient.getTagsNextSinglePageAsync(nextLink, context)
+                .map(res -> Utils.getPagedResponseWithContinuationToken(res, this::getTagProperties));
+        } catch (RuntimeException e) {
+            return monoError(logger, e);
+        }
+    }
+
+    /**
+     * Update the attribute identified by `name` where `reference` is the name of the repository.
```

Comment:
nit: since javadoc doesn't highlight backticks, you can just use single quotes instead.